### PR TITLE
1856: Shares take one turn around the table to be sellable after bought

### PR DESCRIPTION
**Pull request by ventusignis**

### CODE REVIEW

File: lib/engine/step/g_1856/buy_sell_par_shares.rb [hunk 1]
```diff
@@ -22,6 +22,20 @@ def attempt_cgr_action_while_not_floated?(bundle)
...
+          # If the player will be left with at least 1 share, or is fully vested, this is fair game
+          return true unless @round.players_unvested_holdings[player] == bundle.corporation
+
+          # The player has an unvested share, will they be left with at least 1 share?
+          bundle.num_shares < player.num_shares_of(bundle.corporation)
+        end
+
+        def process_buy_shares(action)
+          super
+          puts 'buying shares', action.entity, action.bundle.corporation
```
Comment (jenf) [inline lib/engine/step/g_1856/buy_sell_par_shares.rb:35]: remove puts

Comment (ventusignis) [reply lib/engine/step/g_1856/buy_sell_par_shares.rb]: Thanks, fixed it

File: lib/engine/round/g_1856/stock.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  module Round
+    module G1856
+      class Stock < Stock
+        def players_unvested_holdings
+          # Player -> Corp if player has an unvested share in it.
+          # This doesn't distinguish the number of shares bought. This is fine for 1856
+          #  because the only way to buy more than 1 share is to IPO a corporation and
+          #  if you IPO a corporation you cannot in the same action get rid of the presidency
+          #  so this works. If 1862EA wanted to borrow from this they would have to rework it
+          @players_unvested_holdings ||= {}
```
Comment (tobymao) [inline lib/engine/round/g_1856/stock.rb:15]: line 16 is unnecessary

Comment (ventusignis) [reply lib/engine/round/g_1856/stock.rb:15]: Thanks, fixed

**Pull request merged**